scripts/publish-circleci.js

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env node
+
+const shell = require('shelljs');
+shell.set('-e');
+shell.set('+v');
+
+/*
+ * Assumptions:
+ * 0. You have shelljs installed globally using `npm install -g shelljs`.
+ * 1. The script is running locally - it's not optimized for Travis workflow
+ *    yet.
+ * 2. The script is running in the right branch (e.g., release/vxx.y.z)
+ *
+ * Instructions:
+ * Run this script with SALESFORCEDX_VSCODE_VERSION, CIRCLECI_TOKEN & CIRCLECI_BUILD as environment variables
+ * i.e. SALESFORCEDX_VSCODE_VERSION=x.y.z ./scripts/publish.js
+ *
+ */
+
+// Checks that you are running this with Node v12.4.0 and above
+const [version, major, minor, patch] = process.version.match(
+  /^v(\d+)\.?(\d+)\.?(\*|\d+)$/
+);
+if (parseInt(major) !== 12 || parseInt(minor) < 4) {
+  console.log(
+    'You do not have the right version of node. We require version 12.4.0.'
+  );
+  process.exit(-1);
+}
+// Check if you have installed all the required tooling
+if (!shell.which('vsce')) {
+  console.log('vsce is not installed or could not be found');
+  process.exit(-1);
+}
+
+if (!shell.which('lerna')) {
+  console.log('lerna is not installed or could not be found');
+  process.exit(-1);
+}
+
+if (!shell.which('aws')) {
+  console.log('aws cli is not installed or could not be found');
+  process.exit(-1);
+}
+
+// Checks that you have access to our bucket on AWS
+const awsExitCode = shell.exec(
+  'aws s3 ls s3://dfc-data-production/media/vscode/SHA256.md',
+  {
+    silent: true
+  }
+).code;
+if (awsExitCode !== 0) {
+  console.log(
+    'You do not have the s3 command line installed or you do not have access to the aws s3 bucket.'
+  );
+  process.exit(-1);
+}
+
+// Checks that you have access to the salesforce publisher
+const publishers = shell
+  .exec('vsce ls-publishers', { silent: true })
+  .stdout.trim();
+if (!publishers.includes('salesforce')) {
+  console.log(
+    'You do not have the vsce command line installed or you do not have access to the salesforce publisher id as part of vsce.'
+  );
+  process.exit(-1);
+}
+
+// Checks that you have specified the next version as an environment variable, and that it's properly formatted.
+const nextVersion = process.env['SALESFORCEDX_VSCODE_VERSION'];
+const releaseBranchName = `release/v${nextVersion}`;
+if (!nextVersion) {
+  console.log(
+    'You must specify the next version of the extension by setting SALESFORCEDX_VSCODE_VERSION as an environment variable.'
+  );
+  process.exit(-1);
+} else {
+  const [version, major, minor, patch] = nextVersion.match(
+    /^(\d+)\.(\d+)\.(\d+)$/
+  );
+  const currentBranch = shell
+    .exec('git rev-parse --abbrev-ref HEAD', {
+      silent: true
+    })
+    .stdout.trim();
+
+  if (!currentBranch !== releaseBranchName) {
```

Comment:
Is this supposed to be currentBranch !== releaseBranchName?

Reply:
Yeah, good catch